internal/remote/client.go

```diff
@@ -647,6 +649,35 @@ func (c *Client) maybeCreate(obj model.K8sLocalObject, opts SyncOptions) (*updat
 	return result, nil
 }
 
+func (c *Client) doRecreate(obj model.K8sLocalObject, opts SyncOptions) (*updateResult, error) {
+	ri, err := c.resourceInterfaceWithDefaultNs(obj.GroupVersionKind(), obj.GetNamespace())
+	if err != nil {
+		return nil, errors.Wrap(err, "get resource interface")
+	}
+
+	sio.Debugln("delete " + c.DisplayName(obj))
+	pp := metav1.DeletePropagationForeground
+	err = ri.Delete(obj.GetName(), &metav1.DeleteOptions{PropagationPolicy: &pp})
+	if err != nil && !apiErrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	watcher, err := ri.Watch(metav1.ListOptions{})
```

Comment:
I don't think the watch algo is correct. You are watching on all objects of that type in the namespace and stopping the watch without checking on the name. If a different object gets deleted, it can trigger the end of your watch

Reply:
Sure, solved by `fieldSelector`:

https://github.com/splunk/qbec/blob/c4ceba0b36ce59ce4d0229465a66574236d3f4ff/internal/remote/client.go#L669-L672

crazy bug